dbms/src/Storages/MergeTree/ReplicatedMergeTreeTableMetadata.cpp

```diff
@@ -229,6 +230,21 @@ ReplicatedMergeTreeTableMetadata::checkAndFindDiff(const ReplicatedMergeTreeTabl
                     ErrorCodes::METADATA_MISMATCH);
     }
 
+    if (constraints != from_zk.constraints)
```

Comment:
We need to read them and write to them to zookeeper. https://github.com/yandex/ClickHouse/pull/5273/files#diff-1fa58509f012943f66921580cfcd099dR76 , https://github.com/yandex/ClickHouse/pull/5273/files#diff-1fa58509f012943f66921580cfcd099dR111 .

Reply:
Also, new fields have to be written to the end of metadata.